### PR TITLE
Make obs_vector python only

### PR DESCRIPTION
**Pull request by eivindjahren**
Makes obs_vector python only

towards #5367 
## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/observations/obs_vector.py [hunk 1]
```diff
@@ -16,89 +13,58 @@
...
-        for step in self.getStepList():
-            yield self.getNode(step)
+        return (self.nodes[i] for i in self.getStepList())  # type:ignore
 
     def getStepList(self) -> List[int]:
         """
         Will return an IntVector with the active report steps.
         """
-        return _clib.obs_vector_get_step_list(self)
+        return sorted(self.steps)
```
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/observations/obs_vector.py:57]: Is it better to sort on insert? I assume this happens more often than insertion?

File: src/ert/_c_wrappers/enkf/observations/obs_vector.py [hunk 2]
```diff
@@ -16,89 +13,58 @@
...
-        return self.getKey()
+        return self.observation_key
 
     def getKey(self) -> str:
-        return self._get_key()
+        return self.observation_key
 
     def getObsKey(self) -> str:
-        return self._get_obs_key()
+        return self.observation_key
```
Comment (andreas-el) [inline src/ert/_c_wrappers/enkf/observations/obs_vector.py:42]: `getObservationKey`, `getKey` and `getObsKey` all return the same string.

This __repr__ does not make much sense:
```
f"key = {self.getKey()}, obs_key = {self.getObsKey()}, "
```

**Review by andreas-el — CHANGES_REQUESTED** [review]
This seems a bit off

File: src/ert/_c_wrappers/enkf/observations/obs_vector.py (hunk 1 above)
Comment (eivindjahren) [reply src/ert/_c_wrappers/enkf/observations/obs_vector.py]: Changed it to a `SortedList`

File: src/ert/_c_wrappers/enkf/observations/obs_vector.py (hunk 2 above)
Comment (eivindjahren) [reply src/ert/_c_wrappers/enkf/observations/obs_vector.py:43]: Changed the repr to be the same as the constructor. These are weird, but did actually return the same thing. I was thinking to clean up the usage of these in a separate step.

**Review by andreas-el — APPROVED** [review]
Good job! :guitar:

**Pull request merged**